AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -206,13 +207,57 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    private suspend fun findScreenshotFiles(
+        resultPath: GcsPath,
+        testIdentifiers: List<TestRunnerService.TestIdentifier>
+    ): Map< TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>> {
+        val screenshotArtifactsBlobs = mutableMapOf<TestRunnerService.TestIdentifier, MutableList<TestRunnerService.TestCaseArtifact>>()
+        val screenshotArtifacts: Map<TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>>
+        val testNames = testIdentifiers.associateBy { testIdentifier ->
+            "${testIdentifier.className}_${testIdentifier.name}"
+        }
+        val testRunNumber = testIdentifiers.first().runNumber
+        fun BlobVisitor.fullDeviceId() = relativePath.substringBefore('/', "")
+        googleCloudApi.walkEntires(
+            gcsPath = resultPath
+        ).forEach { visitor ->
+            val runNumber = DeviceRun.create(visitor.fullDeviceId()).runNumber
+            if (runNumber == testRunNumber) {
+                val testName = testNames.keys.find { testName ->
+                    visitor.fileName.startsWith(testName)
+                }
+                val testIdentifier = testNames[testName]
+                if (testIdentifier != null) {
+                    screenshotArtifactsBlobs.getOrPut(testIdentifier) {
+                        mutableListOf()
+                    }.add(
+                        TestRunnerService.TestCaseArtifact(
+                            ResultFileResourceImpl(visitor),
+                            visitor.fileName.substringAfterLast(".")
+                        )
+                    )
+                }
+            }
+        }
+        screenshotArtifacts = screenshotArtifactsBlobs
```

Comment:
why not just return `screenshotArtifactsBlobs` and avoid declaring `screenshotArtifacts` ?

Reply:
private methods shouldn't return mutable maps or list, right?
That's why I added another variable screenshotArtifacts. 
We were following the same pattern in findresults method.

Reply:
doesn't matter, you can return a MutableMap<K, V> as Map<K,V> so your return type can still be immutable

Reply:
Updated in the next PR